pemutil/pem.go

```diff
@@ -299,6 +300,16 @@ func ReadCertificate(filename string, opts ...Options) (*x509.Certificate, error
 	}
 }
 
+type noValidPEMCertError struct {
+	filename string
+}
+
+func (e *noValidPEMCertError) Error() string {
+	return fmt.Sprintf("file %s does not contain a valid PEM formatted certificate", e.filename)
+}
+
+var ErrNoValidPEMCert *noValidPEMCertError
```

Comment:
Something like can be more generic:
```suggestion
type invalidPEMFileError string

func (e invalidPEMFileError) Error() string {
	return fmt.Sprintf("file %s does not contain a valid PEM encoded data", string(e))
}

func (e invalidPEMFileError) Is(target error) bool {
	return target == ErrInvalidPEM
}

var ErrInvalidPEM = errors.New("invalid PEM encoded data")
```

You can also use a specific error for certificates if you want, but always with the `Is(error) bool` method, so you can do things like this:
```go
func foo() error {
	return invalidPEMFileError("file.crt")
}

func main() {
	if err := foo(); err != nil && errors.Is(err, ErrInvalidPEM {
		fmt.Fprintln(os.Stderr, "it doesn't look right")
	}
}
```

In some cases, you might also want to make the error public so the user can cast to it and get the `Filename`. But this can be added later. Later, we can generalize that to something more complex, for example things like:

```go
type InvalidPEMError struct {
	Type PEMType
	Filename string
	Reason string
}
```